### PR TITLE
[KBFS-2747] Fix menus on mobile

### PR DESCRIPTION
**Pull request by jzila**
This is ugly, but it mostly works.

screen shot 2018-02-21 at 6 06 16 pm

screen shot 2018-02-21 at 6 06 28 pm

screen shot 2018-02-21 at 6 06 39 pm

screen shot 2018-02-21 at 6 06 54 pm

screen shot 2018-02-21 at 6 07 19 pm

screen shot 2018-02-21 at 6 07 32 pm

### CODE REVIEW

**Comment by chrisnojima** [comment]
I thought maybe we didn't even have menus for the breadcrumbs on mobile?

**Comment by jzila** [comment]
I wanted to make it function first, and then worry about converging to the design later.

File: shared/common-adapters/popup-menu.desktop.js [hunk 1]
```diff
@@ -99,7 +99,7 @@ class PopupMenu extends Component<Props> {
             e.stopPropagation()
           }}
         >
-          <ModalLessPopupMenu {...(this.props: Props)} />
+          <ModalLessPopupMenu {...(this.props: ModalLessPopupMenuProps)} />
```
Comment (chrisnojima) [inline shared/common-adapters/popup-menu.desktop.js]: this type of typing isn't necessary, just do {...this.props}

File: shared/fs/header/breadcrumb-popup.js [hunk 2]
```diff
@@ -28,14 +29,13 @@ const BreadcrumbPopupMenu = ({routeProps}: PopupMenuProps) => {
   const isTeamPath = routeProps.get('isTeamPath')
   const items = routeProps.get('items')
   const onOpenBreadcrumb = routeProps.get('onOpenBreadcrumb')
+  const onHidden = routeProps.get('onHidden')
   const popupItems = items.map((i, idx) => ({
-    onClick: () => {
-      onOpenBreadcrumb(i.path)
-    },
+    onClick: onOpenBreadcrumb(i.path),
```
Comment (chrisnojima) [inline shared/fs/header/breadcrumb-popup.js]: we rarely do something like this. why make a function return a function here

File: shared/fs/header/header-container.js [hunk 3]
```diff
@@ -2,41 +2,57 @@
...
+  ) => (evt?: SyntheticEvent<>) => void,
+}
+
 const mapStateToProps = () => ({})
 
 const mapDispatchToProps = (dispatch: Dispatch) => ({
-  onOpenBreadcrumb: (path: string) =>
-    dispatch(navigateTo([fsTab, {props: {path: Types.stringToPath(path)}, selected: 'folder'}])),
-  onOpenBreadcrumbDropdown: (
+  onOpenBreadcrumb: (path: string) => (evt?: SyntheticEvent<>) => {
```
Comment (chrisnojima) [inline shared/fs/header/header-container.js]: lets not do this. just use mergeProps to export a function which takes the evt and passes that down into dispatchProps_onOpenBreadcrumb

Comment (jzila) [reply shared/fs/header/header-container.js]: Sure, but isn't that effectively the same thing?

Comment (chrisnojima) [reply shared/fs/header/header-container.js]: no. one is a function we're exporting and another is a function that you call that gives you another function. we don't use this pattern anywhere

File: shared/fs/header/breadcrumb-popup.js (hunk 2 above)
Comment (jzila) [reply shared/fs/header/breadcrumb-popup.js]: OK so we do this in order to support both the breadcrumbs in the dropdown and in the top-level menu with the same function.

Comment (jzila) [reply shared/fs/header/breadcrumb-popup.js]: Actually nvmind I think I can do this your way too.

File: shared/fs/header/header-container.js [hunk 4]
```diff
@@ -11,26 +11,23 @@ type OwnProps = {
...
+  _onOpenBreadcrumb: (path: string, evt?: SyntheticEvent<>) => {
     dispatch(navigateTo([fsTab, {props: {path: Types.stringToPath(path)}, selected: 'folder'}]))
     evt && evt.stopPropagation()
   },
-  _onOpenBreadcrumbDropdown: (
-    dropdownItems: Array<Types.PathBreadcrumbItem>,
-    isTeamPath: boolean,
-    onOpenBreadcrumb: (path: string) => (evt?: SyntheticEvent<>) => void
-  ) => (evt?: SyntheticEvent<>) =>
+  _onOpenBreadcrumbDropdown: (dropdownItems: Array<Types.PathBreadcrumbItem>, isTeamPath: boolean) => (
```
Comment (chrisnojima) [inline shared/fs/header/header-container.js]: the second arrow is incorrect right

Comment (jzila) [reply shared/fs/header/header-container.js]: Right. I'll fix. I also missed fixing the same pattern in the sortbar, so I'll do that too.

File: shared/fs/header/breadcrumb-popup.js [hunk 5]
```diff
@@ -1,7 +1,7 @@
 // @flow
 import * as React from 'react'
 import * as Types from '../../constants/types/fs'
-import {ModalLessPopupMenu} from '../../common-adapters/popup-menu'
+import PopupMenu from '../../common-adapters/popup-menu'
```
Comment (songgao) [inline shared/fs/header/breadcrumb-popup.js]: Looks like we are still using `PopupMenu` for now?

**Comment by jzila** [comment]
@chrisnojima: ready for another look.

File: shared/fs/header/breadcrumb-popup.js (hunk 5 above)
Comment (jzila) [reply shared/fs/header/breadcrumb-popup.js]: Yeah until we figure out a better way to do it.

Comment (chrisnojima) [reply shared/fs/header/breadcrumb-popup.js]: Let's leave these classes alone for now. We do need to fix them up but let's deal with that separately

**Pull request merged**